yarn.config.cjs

```diff
@@ -379,8 +379,10 @@ async function workspaceFileExists(workspace, path) {
 function expectWorkspaceField(workspace, fieldName, expectedValue = undefined) {
   const fieldValue = get(workspace.manifest, fieldName);
 
-  if (expectedValue) {
+  if (expectedValue !== undefined && expectedValue !== null) {
     workspace.set(fieldName, expectedValue);
+  } else if (expectedValue === null) {
+    workspace.unset(fieldName);
   } else if (fieldValue === undefined || fieldValue === null) {
```

Comment:
It looks like the following is implied by the `else`. What do you think about stating it explicitly? It would be redundant, but it would also make it easier for readers to reason about the logic here.

```suggestion
  } else if (expectedValue === undefined && (fieldValue === undefined || fieldValue === null)) {
```

Reply:
Sure, done! 25ef1b75159c422f3e23773d86053b2a51a19e39